### DISCUSSION TITLE
【收集贴】安装兰空 v2 常见问题及其解决方法

### DISCUSSION THREAD

**Discussion by akatsukiro** [lsky-org/lsky-pro#357] [教程]
应该没有其他情况了，不更了。

##  先看官方文档

**[官方文档-常见问题](https://docs.lsky.pro/docs/v2/quick-start/questions.html)**

## 反代/套CDN后无法加载静态文件(css, js)（以及安装时点击无反应）

参考 #317

编辑 .env 文件，追加一行：
```
ASSET_URL=https://你的域名
```
但是接口域名还是 http 的，所以在这个文件的 boot 方法里，
https://github.com/lsky-org/lsky-pro/blob/6d0631b494c31788396cb948f691fe542e5bf8b4/app/Providers/AppServiceProvider.php#L31-L52
第 32 行下面追加一行：
```
\Illuminate\Support\Facades\URL::forceScheme('https');
```

### 方法一： `自定义Web图床` 插件

参考 #346

按如下方式填写即可：
- API地址：https://your.domain/api/v1/upload
- POST 参数名：file
- JSON路径：data.links.url
- 自定义请求头：{"Authorization": "Bearer your_token"}
- 其他选填

### 方法二：`lankong` 插件

参考：https://github.com/hellodk34/picgo-plugin-lankong

按如下方式填写：
- 将 `Lsky Pro Version` 选择为 V2
- Server： 填写你的图床地址即可，**不需要 `/api/v1/upload`** ，**不要以 `/` 结尾**
- Auth Token： `Bearer Your_token`
- 其他选填

## 对接ShareX

https://github.com/lsky-org/lsky-pro/discussions/357#discussioncomment-2535721

Update: 根据 https://github.com/lsky-org/lsky-pro/discussions/357#discussioncomment-4122083 ，新版 ShareX 需注意转义符。

如图。

![image](https://user-images.githubusercontent.com/69096852/162601336-893f7ac4-f4ed-4e5a-8547-3a709eea2b03.png)

![image](https://user-images.githubusercontent.com/69096852/162599598-3fb71de2-7282-41c5-a801-a7f8f437365e.png)

需要指定上传的存储策略的话在 `URL参数` 中添加
| 名称 | 值 |
| --- | --- |
| strategy_id | 1（对应的存储策略D） |

### 剪切板/文件导入

- 未指定存储策略
  ```
  {
    "Version": "13.7.0",
    "Name": "Lsky Pro v2",
    "DestinationType": "ImageUploader",
    "RequestMethod": "POST",
    "RequestURL": "https://lsky.pro/api/v1/upload",
    "Headers": {
      "Authorization": "Bearer Your_token"
    },
    "Body": "MultipartFormData",
    "FileFormName": "file",
    "URL": "$json:data.links.url$"
  }
  ```
- 指定存储策略
```
  {
    "Version": "13.7.0",
    "Name": "Lsky Pro v2",
    "DestinationType": "ImageUploader",
    "RequestMethod": "POST",
    "RequestURL": "https://lsky.pro/api/v1/upload",
    "Parameters": {
      "strategy_id": "2（对应的存储策略id）"
    },
    "Headers": {
      "Authorization": "Bearer Your_Token"
    },
    "Body": "MultipartFormData",
    "FileFormName": "file",
    "URL": "$json:data.links.url$"
  }
```

## 获取Token

参考 #336

### CURL

- 简洁一点：

    ```
    curl -X POST -F "email=your_email@address" -F "password=your_passwd" https://your.domain/api/v1/tokens
    ```
- 展开：
    ```
    curl --location --request POST 'https://your.domain/api/v1/tokens' \
    --form 'email="your_email@address"' \
    --form 'password="your_passwd"'
    ```

### Shell wget
```
wget --no-check-certificate --quiet \
  --method POST \
  --timeout=0 \
  --header '' \
  --body-data 'email=your email address&password=password' \
   'https://pic.iqy.ink/api/v1/tokens'
```

### Postman/ApiPost

![postman](https://user-images.githubusercontent.com/22728201/159217649-32fdd9bc-e84f-422b-bc9c-97a9288d5672.png)

![apipost](https://user-images.githubusercontent.com/69096852/160063787-d93bb84f-1fb2-4c31-93b7-21d886ce0d9d.png)

## 修改底部版权信息

参考 #337

修改位置：

https://github.com/lsky-org/lsky-pro/blob/b8471f2ca25142e5f91d5d0070b46f5277cb275c/resources/views/welcome.blade.php#L33

## 站点无法加载

出现类似下方的报错：
```
Warning: require(/www/wwwroot/your.domain/public/../vendor/autoload.php): Failed to open stream: No such file or directory in /www/wwwroot/your.domain/public/index.php on line 34

Fatal error: Uncaught Error: Failed opening required '/www/wwwroot/your.domain/public/../vendor/autoload.php' (include_path='.:') in /www/wwwroot/your.domain/public/index.php:34 Stack trace: #0 {main} thrown in /www/wwwroot/your.domain/public/index.php on line 34
```

说明你下载错文件了，你下的是源代码。请到[Release](https://github.com/lsky-org/lsky-pro/releases) 页面下载程序。
一般来说命名规则为 `lsky-pro-2.x.x.zip` 。

### 使用MySQL/MariaDB 时报错

```
Dropped all tables successfully.
Migrating: 2014_10 _10_000000_create._groups_table
Installation error!

SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near "json not null comment '组配置'，`created_at` timestamp null, `updated_at` tim' at line 1 (SQL: create table `groups` ( `id` bigint unsigned not null auto_increment primary key, `name` varchar(64) not null comment '角色组名称'，`is_default` tinyint(1} not null default '0' comment '是否默认', 'is_guest' tinyint(1) not null default '0' comment '是否为游客组', 'configs' json not null comment '组配置'，`created. at` timestamp null, 'updated_at' timestamp null) default character set utf8mb4 collate 'utf8mb4_ unicode ci' engine = InnoDB)
```
请使用MySQL 5.7/MariaDB 10.2 及以上版本。

### 使用sqlite时出错

```
SQLSTATE[HY000]: General error: 1 no such function: json (SQL: update "strategies" set "updated_at" = YYYY-MM-DD hh:mm:ss, "configs" = json_patch(ifnull("configs", json('{}')), json({"url":"https:\/\/your.domain\/i"})))
```
请升级sqlite。
Debian / Ubuntu：
```
sudo apt upgrade sqlite3
```

**Comment by 0xxb** [comment]
点个赞 👍

**Comment by imGmc** [comment]
经测试，只能选择MySQL5.7版本安装，MariaDB10版本无法安装:cold_sweat:

**Reply by akatsukiro** [reply]
MariaDB 10.0.x 10.1.x 约等于 MySQL 5.6，你觉得能不能安装呢？

**Reply by louiejordan** [reply]
MariaDB 10.2是对标MySQL5.7的

**Reply by akatsukiro** [reply]
对标，但可惜不是。MariaDB不是MySQL的分支，谢谢。
忘说了，我测试过。MariaDB 10.2是可以正常安装的。

**Reply by imGmc** [reply]
> MariaDB 10.0.x 10.1.x 约等于 MySQL 5.6，你觉得能不能安装呢？

好吧，谢谢😰

**Comment by cdn0x12** [comment]
作者可以补充一下ShareX的对接教程吗，感谢

Update: 根据交流群里群友的指点，ShareX可以参考下面的这个配置（红框内为要注意的部分）

[![](https://i.imgur.com/rIXNJBWm.jpg)](https://i.imgur.com/rIXNJBW.png)

或者[参考这个文件配置](https://pastebin.com/Gz8pBP93)，修改对应内容，保存为文本文件然后导入

**Reply by akatsukiro** [reply]
已补充，谢谢。

**Comment by xiaoyv404** [comment]
很棒的教程

**Reply by akatsukiro** [reply]
谢谢。

**Comment by YuGer26** [comment]
可以补充下第三方存储的图片样式使用方式吗

**Reply by akatsukiro** [reply]
没能理解你的意思。
如果是对接其他存储的话，在图床添加存储策略的地方写的非常清楚了。
如果出现了上传失败的问题的话，请参考 [官方文档](https://docs.lsky.pro/docs/v2/storage/faq.html#%E6%9C%AC%E5%9C%B0%E5%82%A8%E5%AD%98) 检查权限是否正确。
如果是想要自己添加新的官方暂未支持的存储策略，参考该 [官方文档](https://docs.lsky.pro/docs/v2/storage/intro.html#%E5%A6%82%E4%BD%95%E8%AE%BE%E7%BD%AE%E5%BD%92%E5%B1%9E%E8%A7%92%E8%89%B2%E7%BB%84) 。
如果你说的不是以上三种情况之一或全部，请详细说明你的想法。

**Reply by YuGer26** [reply]
官方文档不建议使用自带的原图保护及图片水印，推荐使用第三方存储的；在实际使用中，如果在第三方存储服务中设置了原图保护及水印，接口或者web上传给出得links.url地址并无法直接访问需要加上第三方存储的样式标符（例：?style=xxxx），是否在后台可以直接增加这项配置。

**Reply by YuGer26** [reply]
然后在上传后直接给出加上样式的地址，使得可以直接访问

**Reply by akatsukiro** [reply]
我觉得是不可以的。但是你可以作公告以说明。

**Reply by YuGer26** [reply]
配置第三方设置中的访问域名：http://xxx.xx.com{{图片名称}}?style=xxxx

**Comment by meetcode** [comment]
环境都符合，安装提示成功，最后访问页面提示：目前无法处理此请求。HTTP ERROR 500
请问如何解决

**Reply by penndu** [reply]
> 

伪静态配了吗？如果是宝塔的面板，需关闭防跨站！

**Reply by meetcode** [reply]
伪静态配置了，按照文档配置的，不是用的宝塔，用的amh

**Reply by penndu** [reply]
> 伪静态配置了，按照文档配置的，不是用的宝塔，用的amh

非官方技术支持，需要的话加QQ694027118。

**Reply by meetcode** [reply]
谢了，我换宝塔面板试试吧，不行再麻烦您，谢谢

**Reply by penndu** [reply]
> 谢了，我换宝塔面板试试吧，不行再麻烦您，谢谢

友情提示，如果内存小于1G，安装fileinfo扩展可能会失败，添加swap空间即可。

**Reply by meetcode** [reply]
换宝塔了，伪静态都已经设置，防跨域已经关闭，最后提示安装成功，访问页面，还是提示HTTP ERROR 500

**Reply by penndu** [reply]
> 换宝塔了，伪静态都已经设置，防跨域已经关闭，最后提示安装成功，访问页面，还是提示HTTP ERROR 500

复制一下图床源码下载地址。

**Reply by skkshu** [reply]
你好，我也遇到了相同的问题。解决方案：我之前是用wget直接在服务器上下载的，一直报http500的错，后来我尝试先将压缩包下载到windows，再用宝塔上传压缩包，解压到网站目录，没有http500的报错了，希望对后来的人有所帮助。

**Reply by liub1934** [reply]
宝塔面板我用PHP 8.2的版本也是这样的，安装报500，PHP切换成8.1能成功安装

**Reply by liub1934** [reply]
安装正常访问出现500的，可以看下安装目录storage/logs下的日志，如果有下面的类似数据库连接错误，检查下数据库的密码，不要使用特别复杂的含有特殊符号的密码，本人使用AMH面板遇到的，仅供参考。
```
prod.ERROR: SQLSTATE[HY000] [1045] Access denied for user 'xxx'@'localhost' (using password: YES) (SQL: select * from `groups` where `is_guest` = 1 limit 1) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 1045): SQLSTATE[HY000] [1045] Access denied for user 'xxx'@'localhost' (using password: YES) (SQL:  select * from `groups` where `is_guest` = 1 limit 1) at /home/wwwroot/xxx/web/vendor/laravel/framework/src/Illuminate/Database/Connection.php:759)
[stacktrace]
```

**Reply by SANYIQIN** [reply]
同 刚试了半天 发现是PHP版本的问题 感谢指出 也提醒友友们此程序宝塔面板（伪静态+防跨站关闭+运行目录设置PUBLIC）在使用PHP8.2的情况下安装流程报500 切换8.1同样的禁用函数和插件一秒解决

**Reply by wxjback** [reply]
> 安装正常访问出现500的，可以看下安装目录storage/logs下的日志，如果有下面的类似数据库连接错误，检查下数据库的密码，不要使用特别复杂的含有特殊符号的密码，本人使用AMH面板遇到的，仅供参考。
> 
> ```
> prod.ERROR: SQLSTATE[HY000] [1045] Access denied for user 'xxx'@'localhost' (using password: YES) (SQL: select * from `groups` where `is_guest` = 1 limit 1) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 1045): SQLSTATE[HY000] [1045] Access denied for user 'xxx'@'localhost' (using password: YES) (SQL:  select * from `groups` where `is_guest` = 1 limit 1) at /home/wwwroot/xxx/web/vendor/laravel/framework/src/Illuminate/Database/Connection.php:759)
> [stacktrace]
> ```

牛B，就是这个问题，我是 docker 部署的 折腾了一天，能正常安装，就是访问首页报 500，查看 docker 的 mysql 报  access denied， 一开始以为账号密码权限问题，但是排查了一遍都没问题，后来在.evn中改了简单密码，就ok 了，这evn 文件的编码解析是不是有问题，复杂密码 符号，我就用 !@# 这几个都不行。感谢兄弟！

**Comment by zigzhu** [comment]
我用群晖docker部署在本地没有问题，可以正常使用。但是用群晖自带的反向代理后，外网访问就会400报错。
请问如何解决

**Comment by love2wind** [comment]
请问一下，怎么修改文字logo为图片，怎么修改背景，找了一圈没找到怎么改

**Comment by YangYongNET** [comment]
请教一下下面错误怎么解决
SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes (SQL: alter table `password_resets` add index `password_resets_email_index`(`email`))

**Reply by TaoRone** [reply]
同问

**Reply by liub1934** [reply]
将MySQL默认的MyISAM引擎手动在配置文件中改为InnoDB即可
default-storage-engine = InnoDB

**Comment by xxlgenius** [comment]
shareX更新14.0版本之后重写了自定义上传器语法解析器，导致Authorization字段中“|”字符之后的内容会被吞。这是最新的配置文件
[![image](https://user-images.githubusercontent.com/39547011/201458362-2853c05c-0a9c-4843-826e-eedcaeb08d09.png)
](url)

**Reply by louiejordan** [reply]
我说怎么一直上传失败，感谢分享

**Reply by Merlynr** [reply]
牛的，太折磨了，这个错误，转义符

**Reply by suixintianya** [reply]
谢谢你，终于成功了

**Reply by ruinny** [reply]
终于解决了这个问题，全网难题

**Comment by cenlios** [comment]
请问这个怎么解决？
 [core:notice] [pid 13] AH00094: Command line: '/usr/sbin/apache2 -D FOREGROUND'
 [mpm_prefork:notice] [pid 13] AH00163: Apache/2.4.57 (Debian) PHP/8.1.21 configured -- resuming normal operations
[core:warn] [pid 13] AH00098: pid file /var/run/apache2/apache2.pid overwritten -- Unclean shutdown of previous Apache run?
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.8. Set the 'ServerName' directive globally to suppress this message
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.8. Set the 'ServerName' directive globally to suppress this message

**Reply by leoleelxh** [reply]
同问

**Reply by leoleelxh** [reply]
请问解决了吗？

**Reply by leoleelxh** [reply]
自问自答一下，这个是警告可以忽略的，当然也可以在阿帕奇配置文件中，写入ServerName localhost，去除警告。
如果因为更新了docker，导致无法访问webui，把web port改回成现在端口就可以了。
新版：8096。

**Comment by iceyumao** [comment]
上传图片时有一些特殊符号比如'会导致bbcode失效，请问如何修改自动替换这些符号？
之前1.x版本是会自动更换成_

**Comment by woodchen-ink** [comment]
太感谢了，套了CDN一会儿js,css回源http，一会儿回源https，官方文档里没找到说明，人都麻了，还好google到了这个文章。

**Comment by labspc** [comment]
主机环境 macOS 14.2，客户端使用 PicGo -2.4.0-beta.1，插件使用 lankong 1.1.3，在 V2 模式下按照上述方法二正确设置后，仍然无法使用 PicGo进行上传。但是，在网页测试，可以正常上传。

**Reply by akatsukiro** [reply]
方便展示一下您的配置信息吗？

**Reply by labspc** [reply]
反馈，在上述环境按照方法二正确设置，测试成功。如果按照教程设置后，仍无法上传，可参考：
https://github.com/lsky-org/lsky-pro/discussions/357#discussioncomment-8316819

**Comment by labspc** [comment]
<img width="365" alt="截屏2024-01-31 16 44 19" src="https://github.com/lsky-org/lsky-pro/assets/47222768/d57d224d-24a9-4113-9896-59b5c94dc272">

Token 是自动生成的，不需要 POST，后台是兰空企业版。

**Reply by akatsukiro** [reply]
不需要添加 `/api/v1/upload` ，您似乎是添加了这部分？

**Reply by labspc** [reply]
server 栏没有就是 domain，没有添加别的，也试过添加 /upload ，还有 V2、V1，Bearer 后加 \ 都试过了。目前，不知道错误在哪里。PicGo上传时，只显示进度条红色，就是上传不了。

**Reply by labspc** [reply]
是否是 PicGo，版本过高的问题？ 其余设置均按照教程来的，没有问题。

**Reply by labspc** [reply]
已解决。🤝   道友们，可以看看自己服务器是否有拦截 PicGo 的上传请求，解除拦截即可成功上传。

**Comment by flycat233** [comment]
配置的阿里云OSS，上传图片会提示上传失败，图片上传失败
![Uploading {B97A3878-3F4A-4ed2-8915-04CF10D23E53}.png…]()

**Comment by RealHeart** [comment]
修改 gravatar 镜像地址

> https://github.com/lsky-org/lsky-pro/blob/911275c13b038c7a8b710de44664f23887eeb6f6/app/Utils.php#L33-L39

35 行

**Comment by RuSheng920** [comment]
用的alist接入的储存策略，上传图片后，复制链接打开显示这个
{
“code”: 401,
“message”: “expire missing”,
“data”: null
}
![Uploading ef325663cfe00ba71459c0730e6e6e7d.png…]()

**Reply by 2982136527** [reply]
佬有解决吗，我也遇到了

**Comment by Tianlengle** [comment]
群辉docker环境安装  卡在最后一一步写入配置数据库 报错
Dropping all tables ......................................... 120,096ms FAIL
Installation error!

SQLSTATE[HY000] [2002] Connection timed out (SQL: SHOW FULL TABLES WHERE table_type = 'BASE TABLE')
请问有大佬遇到过吗

**Reply by abloom25** [reply]
如果你用的是网址数据库连接那块填公网ip，
确保数据库可以在公网访问，
我以前用1p忘了打开mysql的端口外部访问，
导致了这个问题。

**Reply by licoba** [reply]
> 如果你用的是网址数据库连接那块填公网ip， 确保数据库可以在公网访问， 我以前用1p忘了打开mysql的端口外部访问， 导致了这个问题。

没必要吧，还要走一遍公网，明显内网连接比较快啊

**Reply by licoba** [reply]
> 群辉docker环境安装 卡在最后一一步写入配置数据库 报错 Dropping all tables ......................................... 120,096ms FAIL Installation error!
> 
> SQLSTATE[HY000] [2002] Connection timed out (SQL: SHOW FULL TABLES WHERE table_type = 'BASE TABLE') 请问有大佬遇到过吗

https://www.nodeseek.com/post-186670-1  看这里

**Reply by abloom25** [reply]
> > 如果你用的是网址数据库连接那块填公网ip， 确保数据库可以在公网访问， 我以前用1p忘了打开mysql的端口外部访问， 导致了这个问题。
> 
> 没必要吧，还要走一遍公网，明显内网连接比较快啊

哈哈，那就不清楚了。反正我是这么解决的。

**Comment by xiaojinmi** [comment]
Artisan 安装命令执行失败，请检查环境配置或日志信息。咋办啊，install.sh安装失败
